src/main/java/org/humancellatlas/ingest/project/ProjectService.java

```diff
@@ -60,7 +64,17 @@ public Project addProjectToSubmissionEnvelope(SubmissionEnvelope submissionEnvel
                                      });
     }
 
-    public Page<Project> queryByContent(List<MetadataCriteria> query, Pageable pageable){
-        return this.projectRepository.findByContent(query, pageable);
+    public Page<Project> queryByContent(List<MetadataCriteria> query, Optional<Boolean> isUpdate, Pageable pageable){
+        return this.projectRepository.findByContent(query, isUpdate, pageable);
+    }
+
+    public Page<SubmissionEnvelope> getProjectSubmissionEnvelopes(Project project, Pageable pageable) {
+        Page<Project> projects = this.projectRepository.findByUuid(project.getUuid(), Pageable.unpaged());
+        List<SubmissionEnvelope> envelopes = new ArrayList<>();
+        for (Project projectDocument : projects)
```

Comment:
findByUuid is unlikely to return multi page output in most cases, but still we should maybe not assume that and make sure all projects are iterated over, not just in this page. stream?